packages/genesys-spark-components/src/components/stable/gux-datepicker/gux-datepicker.tsx

```diff
@@ -542,23 +545,53 @@ export class GuxDatepicker {
   }
 
   updateDate() {
-    if (this.mode === CalendarModes.Range) {
-      const [from, to] = fromIsoDateRange(this.value);
-      const { map: map1, regexp: regexp1 } = this.getMapAndRegexFromField(from);
-      this.formattedValue = this.format.replace(regexp1, match => {
-        return map1[match] as string;
-      });
-      const { map: map2, regexp: regexp2 } = this.getMapAndRegexFromField(to);
-      this.toFormattedValue = this.format.replace(regexp2, match => {
-        return map2[match] as string;
-      });
+    if (readRegionalDatesCookie() && !this.formatOverride) {
+      if (this.mode === CalendarModes.Range) {
+        const [from, to] = fromIsoDateRange(this.value);
+        this.formattedValue = sparkIntl
+          .dateTimeFormat(sparkIntl.determineDisplayLocale(this.root), {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric'
+          })
+          .format(new Date(from));
+        this.toFormattedValue = sparkIntl
+          .dateTimeFormat(sparkIntl.determineDisplayLocale(this.root), {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric'
+          })
+          .format(new Date(to));
+      } else {
+        const dateValue = fromIsoDate(this.value);
+        this.formattedValue = sparkIntl
+          .dateTimeFormat(sparkIntl.determineDisplayLocale(this.root), {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric'
+          })
+          .format(new Date(dateValue));
+      }
```

Comment:
Would this be more readable as, I have not tested it fully so there may be errors,

```
if(regionalDatesCookie() && !this.formatOvveride {
   const locale = sparkIntl.determineDisplayLocale(this.root);
  const options = {
  year: 'numeric',
 month: 'numeric',
 day: 'numeric',
}

const sparkIntlFormatter = sparkIntl.dateTimeFormat(locale, options);

if(this.mode === CalendarModes.Range) {
 const [from, to] = fromIsoDateRange(this.value);
const fromDate = new Date(from);
const toDate = new Date(to);

this.formattedValue = sparkIntlFormatter.format(fromDate);
this.toFormattedValue = sparkIntlFormatter.format(toDate);
} else {
 const dateValue = fromIsoDate(this.value);
const formatDate = sparkIntlFormatter.format(new Date(dateValue));
this.formattedValue = formattedDate;

}
}
```